src/Altinn.Platform/Altinn.Platform.Storage/Storage/Controllers/MessageboxInstancesController.cs

```diff
@@ -142,41 +134,64 @@ public class MessageBoxInstancesController : ControllerBase
                 return StatusCode(500, queryResponse.Exception);
             }
 
-            if (queryResponse == null || queryResponse.Count <= 0)
-            {
-                return Ok(new List<MessageBoxInstance>());
-            }
-
-            List<Instance> allInstances = queryResponse.Instances;
-            await RemoveHiddenInstances(allInstances);
+            return await ProcessQueryResponse(queryResponse, language);
```

Comment:
Codesmell line 100, i guess we could only check for includeActive == true, cause we always want this to be false if an archiveref is involved.

Reply:
Then we can set includeActive = false for this block (and drop the check mentioned)?